### ISSUE TITLE
Idea: Easy upgrade from WETH9

### ISSUE THREAD

**Issue by dmihal** [WETH10/WETH10#48]
Any thoughts about a function that would allow easy, cheap conversion from WETH9 into WETH10? Maybe reverse too?

**Comment by moodysalem** [comment]
WETH9 -> WETH10 I see no reason to put in WETH10 vs. external contract

WETH10 -> WETH9 might be nice to avoid extra approvals if you have to use the old WETH and only have the new one

**Comment by albertocuestacanada** [comment]
Unfortunately, [WETH9](https://github.com/makerdao/sai/blob/8b7a7359f40231131218b594fa59ac2bcee5f6ef/src/weth9.sol#L38) doesn't allow withdrawing from any account other than the caller (something I like of WETH10).

Otherwise you could add this in WETH10:
```
    function depositFromContract(address token, uint256 value) external payable {
        WETH9(token).withdrawFrom(msg.sender, value);
    }
```

**Comment by albertocuestacanada** [comment]
WETH10 to WETH9 is easy:
```
    function convert(address token, uint256 value) external {
        withdrawTo(token, value);
        require(IERC20(token).transfer(msg.sender, value), "WETH10::convert: Failed transfer");
    }
```

There are two ways around it, either by passing the WETH9 address in the constructor, or allowing arbitrary ERC20 tokens.
a) Address in constructor is robust, but we need to add a constructor just for this.
b) Address in function gives users a lot of room to give the wrong address and lose their money.

I'm not that bothered about given users the option to downgrade, given that most platforms user Ether in the UI, and wrap and unwrap every time. Holding WETH seems to be the kind of thing only done by those that interact with the mainnet bypassing UIs.

**Comment by albertocuestacanada** [comment]
Fixed in #67, I thought this would drive adoption by reducing the risk of adopting WETH10, making it cheaper for users to go back to WETH9 if critical mass is not achieved.
